src/components/NewItemForm.jsx

```diff
@@ -36,14 +55,41 @@ const NewItemForm = () => {
   };
 
   const handleDialogClose = (dialogState) => {
-    setAnchorEl(null);
+    handleMenuClose();
+    clearAllErrors();
     dialogState(false);
   };
 
+  const navigate = useNavigate();
   // Collections Form
   const [collectionName, setCollectionName] = useState("");
-  const onSaveCollection = () => {
-    console.log(collectionName);
+  const [collectionError, setCollectionError] = useState(clearError);
+  const onSaveCollection = async () => {
+    if (collectionName === "") {
+      setCollectionError(requiredFieldsError);
+      return;
+    } else {
+      setCollectionError(clearError);
+    }
+
+    try {
+      let response = await postCollections(collectionName);
+      setCollectionName("");
+      handleDialogClose(setCollectionsDialogOpen);
+
+      // Get new vault collection items
+      getVaultCollections();
```

Comment:
On vault creation, the server returns the newly created vault. We could update the state with just that one item. Since we don't have to decrypt the vault collections this is likely fine too.

Reply:
Yes good call out, updated